chore(storybook): align typography elements for VRTs

castastrophe opened the pull request:
## Description

Currently VRTs are leveraging the Typography element for the ignored headings. Although they use the "chromatic-ignore" class, this can cause layout issues when the line-height or font-size changes for those typographical elements, leading to unexpected VRT differences. This PR aims to reduce that risk by isolating type styles for VRT from the design system.

In order for the text color to respond to the background context without using a token value, we need to pass the context data through the Container down to the Heading. To this end I've added the context passthroughs in all the existing uses.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] Expect no changes to VRTs
- [x] Expect headings to always show using "light" mode at the top-level of the testing grids
- [x] Expect headings inside the testing grids to respond to the global color context settings (white in dark mode, black in light mode)
- [x] Expect headings on the docs pages to show using "dark" mode when the dark mode styles are applied

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [x] VRTs have been run and looked at.
- [x] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] ✨ This pull request is ready to merge. ✨

castastrophe commented:
Passing VRT: https://www.chromatic.com/build?appId=64762974a45b8bc5ca1705a2&number=3639

cdransf commented on .storybook/decorators/utilities.js:
```diff
@@ -100,6 +131,12 @@ export const Container = ({
+				...context,
+				globals: {
+					...context.globals ?? {},
+					color: level === 1 ? "light" : context.globals?.color,
```
Is a value of `1` for `level` always `"light"`? That association feels unclear when reading it without existing context.

castastrophe replied:
Generally but this is a good call-out because it's not the case here:  https://pr-3343--spectrum-css.netlify.app/preview/?path=/docs/components-action-button--docs&globals=color:dark

castastrophe replied:
This has now been fixed! cc @jawinn

cdransf reviewed (approved):
LGTM! Just had one minor question. ✨

jawinn reviewed (changes requested):
This looks like the same issue you just linked in [your comment](https://github.com/adobe/spectrum-css/pull/3343#discussion_r1821312725), but wanted to point it out. The color of the headings on some "Docs" pages was the only issue I saw.

PR: Docs, dark
![Screenshot 2024-10-29 at 4 12 55 PM](https://github.com/user-attachments/assets/14ad7e54-db1a-43c5-887d-9588e105da63)

Prod: Docs, dark
![Screenshot 2024-10-29 at 4 12 47 PM](https://github.com/user-attachments/assets/58569639-e86b-4270-9dd1-ab03b0b02541)

castastrophe commented:
> > @cdransf @jawinn I added a new test case for the docs pages too and corrected the error we were seeing. Thanks!
> 
> I see `dark` as fixed. It looks like `darkest` [has the same issue](https://pr-3343--spectrum-css.netlify.app/preview/?path=/docs/components-action-button--docs&globals=color:darkest;testingPreview:!true).

Good call-out @jawinn, updated it to `startsWith("dark")` instead of `=== "dark"`
